ws-server/src/server.rs

```diff
@@ -86,13 +85,17 @@ impl Server {
 
 					if connections.count() >= self.cfg.max_connections as usize {
 						log::warn!("Too many connections. Try again in a while.");
+						connections.add(Box::pin(handshake(socket, HandshakeMode::Reject { status_code: 429 })));
```

Comment:
Why do we have to `add()` the connection here? Wouldn't calling `handshake(socket, HandshakeMode::Reject { status_code: 429 })` be enough here?

Reply:
You'd have to `.await` it, which means you'd block incoming connections while you're sending the response here. It would be fine to `.await`, but you could potentially lock the loop if the receiving end of the socket is unusually slow?

Reply:
Got it, ty.